resources/resolver.go

```diff
@@ -752,17 +787,111 @@ func ResolveConfig(vocabId string, token string) (config *HFConfig,
 	}
 	hfConfig.BosTokenStr = &bosToken
 
-	if hfConfig.EosTokenStr == nil {
-		hfConfig.EosTokenStr = &defaultTkn
+	hfConfigPtr, err := ResolveHFFromResources(resources, hfConfig)
+	if err != nil {
+		return nil, nil, err
 	}
-	if hfConfig.PadTokenStr == nil {
-		hfConfig.PadTokenStr = &defaultTkn
+	hfConfig = *hfConfigPtr
+
+	return &hfConfig, resources, nil
+}
+
+// ResolveHFFromResources
+// Given a set of resources, resolve the HuggingFace configuration.
+// Used to be able to resolve both embedded and local resources.
+func ResolveHFFromResources(resources *Resources, hfConfig HFConfig) (*HFConfig, error) {
```

Comment:
Why are you modifying `hfConfig` this way? Why not just consume `*hfConfig` for `ResolvedHFFromResources`?

```suggestion
	hfConfig, err := ResolveHFFromResources(resources, &hfConfig)
	if err != nil {
		return nil, nil, err

	return hfConfig, resources, nil
}

// ResolveHFFromResources
// Given a set of resources, resolve the HuggingFace configuration.
// Used to be able to resolve both embedded and local resources.
func ResolveHFFromResources(resources *Resources, hfConfig *HFConfig) (*HFConfig, error) {

Reply:
You're right, it does look a little funky here, I changed it such that it follows this format and tested it, looks better

```
func ResolveHFFromResources(resources *Resources, hfConfig *HFConfig) (*HFConfig, error) {

hfConfig, err = ResolveHFFromResources(resources, hfConfig)
	if err != nil {
		return nil, nil, err
	}
```